client/src/core/client/stream/tabs/Comments/Comment/ReportFlow/ReportFlowContainer.tsx

```diff
@@ -25,22 +25,32 @@ const ReportFlowContainer: FunctionComponent<Props> = ({
   const onFormClose = useCallback(() => {
     onClose();
   }, [onClose]);
+
+  let anonymousWithDSA = false;
   if (!viewer) {
-    return null;
+    if (!settings.dsa?.enabled) {
+      return null;
+    } else {
+      anonymousWithDSA = true;
+    }
   }
```

Comment:
What are your thoughts on this logic? For me, it improves read-ability, but I have a feeling I might be in the minority thanks to it relying on `?.` operator.

```
const anonymousWithDSA = !viewer && settings.dsa?.enabled;

if (!settings.dsa?.enabled && !viewer) {
  return null;
}
```